scripts/library.sh

```diff
@@ -487,9 +490,34 @@ function get_knative_base_yaml_source() {
   if [[ ${branch_name} =~ ^release-[0-9\.]+$ ]]; then
     # Get the latest tag name for the current branch, which is likely formatted as v0.5.0
     local tag_name="$(git describe --tags --abbrev=0)"
-    knative_base_yaml_source="https://storage.googleapis.com/knative-releases/@/previous/${tag_name}"
+    # The other repo might not have this tag, so we need to find the first available tag before it.
+    echo "$(find_latest_release_yaml_source ${tag_name} ${repo_name} ${source_name})"
+  else
+    local knative_base_yaml_source="https://storage.googleapis.com/knative-nightly/@/latest"
+    echo "${knative_base_yaml_source/@/${repo_name}}/${source_name}.yaml"
   fi
-  echo "${knative_base_yaml_source}"
+}
+
+# Returns latest release yaml source for the given knative project.
+# Parameters: $1 - git tag name of the current branch, e.g. v0.5.1
+#             $2 - repository name of the given project
+#             $3 - short name of the yaml source file
+function find_latest_release_yaml_source() {
+  local tag_name=$1
+  local repo_name=$2
+  local source_name=$3
+  local major_minor=`echo $tag_name | cut -d. -f-2`
+  local revision=`echo $tag_name | cut -d. -f3`
+  for (( ; revision>=0; revision-- ))
+  do
+    local tag_name="$major_minor.$revision"
+    local yaml_source="https://storage.googleapis.com/knative-releases/${repo_name}/previous/${tag_name}/${source_name}.yaml"
+    local ret_code=$(curl -s -o /dev/null -w "%{http_code}" "${yaml_source}")
```

Comment:
Use `gsutil ls` and parse the output, it's faster and simpler.

Reply:
Done.